enhancements/windows-containers/csi-proxy.md

```diff
@@ -0,0 +1,189 @@
+Persistent storage is a necessary feature for production workloads on Kubernetes, and it is important that this
+functionality continue to work, even with the in-tree drivers being removed.
+
+### User Stories
+
+* As a developer using OpenShift, I want the Windows workloads I create to be able to read and write to storage which
+  persists when pods are removed and re-created.
+
+### Goals
+
+* Windows nodes configured by WMCO fulfill all the pre-requisites to run CSI node drivers.
+
+### Non-Goals
+
+* The Windows CSI storage drivers will not be distributed as part of OpenShift.
+  * At this time, we are not able to ship Windows containers images as all Windows container images are packaged
+with a Windows kernel. Red Hat has a policy to not ship 3rd party kernels for support reasons.
+* Enabling the use of volumes which can be shared between Linux and Windows.
+
+## Proposal
+
+The [windows-machine-config-operator](https://github.com/openshift/windows-machine-config-operator/)(WMCO),
+will ensure that csi-proxy is running on the Windows nodes it configures. This will be done through the normal
+workflow for adding a service to the expected service list.
+
+The csi-proxy binary will built from source and baked into the WMCO image. WMCO will copy the binary to each node at
+configuration time. A new item will be added to the windows-services ConfigMap, instructing the
+windows-instance-config-daemon to ensure that csi-proxy is always running as a Windows service on each node.
+
+With the nodes in this state, users are free to follow upstream documentation to deploy the CSI node drivers
+of their choice on the Windows nodes in the cluster.
+
+No support from Red Hat will be given for issues occurring with the upstream drivers, as they are not being distributed
+as part of OpenShift.
+
+
+### Workflow Description
+
+1. The cluster administrator will identify what CSI storage drivers the cluster requires for its workloads.
+2. Based on the requirements of the workloads, the cluster admin can follow upstream documentation to deploy the CSI
+   node drivers on its Windows nodes.
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+* If there are any security issues present in the CSI driver images deployed by the user, they will not be fixed
+by upgrading OCP. It is up to the user to stay informed and update their images as necessary.
+* A mismatch between the OpenShift CSI drivers and the user deployed Windows CSI drivers could cause undefined
+  behavior. This is described in the [version skew](#version-skew-strategy) section.
+
+
+### Drawbacks
+
+The main drawback to this approach is it puts the onus on the cluster administrator. As they will be required
+to deploy and maintain the Windows CSI DaemonSet themselves, and will not be given support for any issues that may
+come up, which do not specifically deal with the csi-proxy Windows service. This may cause an increased load
```

Comment:
```suggestion
to deploy and maintain the Windows CSI DaemonSet themselves, and will only be given support for issues with the csi-proxy Windows service. This may cause an increased load
```